src/compiler/moduleNameResolver.ts

```diff
@@ -1774,19 +1814,18 @@ function nodeModuleNameResolverWorker(features: NodeResolutionFeatures, moduleNa
         affectingLocations,
         packageJsonInfoCache: cache,
         features,
-        conditions,
+        conditions: conditions ?? emptyArray,
         requestContainingDirectory: containingDirectory,
         reportDiagnostic: diag => void diagnostics.push(diag),
         isConfigLookup,
         candidateIsFromPackageJsonField: false,
     };
-
-    if (traceEnabled && moduleResolutionSupportsPackageJsonExportsAndImports(getEmitModuleResolutionKind(compilerOptions))) {
-        trace(host, Diagnostics.Resolving_in_0_mode_with_conditions_1, features & NodeResolutionFeatures.EsmMode ? "ESM" : "CJS", conditions.map(c => `'${c}'`).join(", "));
+    if (traceEnabled && moduleResolutionSupportsPackageJsonExportsAndImports(moduleResolution)) {

```

Comment:
Is this enough or do we need to check if conditions array is nonempty as well ?

Reply:
This is correct since `default` always matches, rather than being a real condition. Also, imports/exports don’t have to specify conditions, they can just point subpaths to files unconditionally.